api/src/pcapi/utils/custom_logic.py

```diff
@@ -30,31 +32,39 @@ def less(a: object, b: object, *args) -> bool:  # type: ignore [no-untyped-def]
     return a < b and (not args or less(b, *args))  # type: ignore [operator]
 
 
-def less_or_equal(a: object, b: object, *args) -> bool:  # type: ignore [no-untyped-def]
+def less_or_equal(a: object, b: object, *args: typing.Any) -> bool:
     return (less(a, b) or soft_equals(a, b)) and (not args or less_or_equal(b, *args))
 
 
-def contains(a, b) -> bool:  # type: ignore [no-untyped-def]
-    """Checks if the list b contains the element a"""
+def contains(a: str, b: list) -> bool:
```

Comment:
Est-ce plus précisément `list[str]` ?
```suggestion
def contains(a: str, b: list[str]) -> bool:
```